### ISSUE TITLE
Repeated packet sending loop

### ISSUE THREAD

**Issue by kcin2** [GrimAnticheat/Grim#970]
### Describe the bug and how to replicate it

Install a plugin using ProtocolLib that modifies Set_Slot & Window_Items packets in any way.
In this example I set the Material of all items to TNT

This will cause repeated window_items/set_slot packets to be sent per item modified * however many players are on the server.
Showcase: https://www.youtube.com/watch?v=q2T-hZ7NK50

```
public class PacketDebugger extends PacketAdapter implements Listener {

    public PacketDebugger(@NotNull PacketAdapter.AdapterParameteters params) {
        super(params);
    }

    @Override
    public void onPacketSending(PacketEvent event) {
        Player player = event.getPlayer();
        if (player.getGameMode() == GameMode.CREATIVE) return;

        PacketContainer packet = event.getPacket();
        ItemStack item = packet.getItemModifier().read(0);
        if (item != null && !item.getType().isAir()) {
            item.setType(Material.TNT);
        }

        if (event.getPacketType() == PacketType.Play.Server.WINDOW_ITEMS) {
            player.sendMessage("WINDOW_ITEMS");
            List<ItemStack> items = packet.getItemListModifier().read(0);
            for (ItemStack itemStack : items) {
                if (itemStack != null && !itemStack.getType().isAir()) {
                    itemStack.setType(Material.TNT);
                }
            }
        } else {
            player.sendMessage("SET_SLOT");
        }
    }
}

### Grim version

Latest Spigot / 2.3.36

### Server version

git-Paper-448 (MC: 1.19.3) (Implementing API version 1.19.3-R0.1-SNAPSHOT) (Git: 155aa36)


### Plugins

GrimAC, ProtocolLib, and the test plugin

**Comment by MWHunter** [comment]
Yeah, grim is detecting a desync in the client's inventory due to this logic... the logic isn't designed for when you purposefully create ghost items.

**Comment by kcin2** [comment]
I figured as much; what do you think the best way to fix/mitigate this is?

Not sure what checks heavily rely on that system
Please correct me if I'm wrong, but I assume migrating this plugin to using PE wouldn't help anything in this regard

In short, I currently use this system in my plugin with ProtocolLib to edit items as they're sent out to change a set of select things based on the players settings, such as coloring of certain attributes, etc. I also remove some data I don't want players to be able see by viewing the NBT, such as item origin/tracking information that's stored on the item.

**Comment by MachineBreaker** [comment]
> I figured as much; what do you think the best way to fix/mitigate this is?
> 
> Not sure what checks heavily rely on that system Please correct me if I'm wrong, but I assume migrating this plugin to using PE wouldn't help anything in this regard
> 
> In short, I currently use this system in my plugin with ProtocolLib to edit items as they're sent out to change a set of select things based on the players settings, such as coloring of certain attributes, etc. I also remove some data I don't want players to be able see by viewing the NBT, such as item origin/tracking information that's stored on the item.

Does this actually cause you any problem? I'll try to fix this when i have time, if possible can you send me an compiled plugin that actually debug this issue?

**Comment by kcin2** [comment]
To my knowledge grim functions as intended. So does the plugin as you can see from the video. It's just a matter of reprocessing any changes to items in my plugin, as well as the potentially massive amount of constant unnecessary window_items packets being sent. 

Yes I can send that, MachineBreaker#2242 I assume

**Comment by MachineBreaker** [comment]
> To my knowledge grim functions as intended. So does the plugin as you can see from the video. It's just a matter of reprocessing any changes to items in my plugin, as well as the potentially massive amount of constant unnecessary window_items packets being sent.
> 
> Yes I can send that, MachineBreaker#2242 I assume

Yes

**Comment by MWHunter** [comment]
My solution would be to not check against bukkit after the server sets a slot.  There's no need to do that since it is the known good server state - and the current behaviour of doing this is unintentional

**Comment by MachineBreaker** [comment]
> My solution would be to not check against bukkit after the server sets a slot. There's no need to do that since it is the known good server state - and the current behaviour of doing this is unintentional

I'll try to do this
